cmd/tdex/depositfee.go

```diff
@@ -27,17 +55,306 @@ func depositFeeAction(ctx *cli.Context) error {
+			}
+			unspents = append(unspents, u...)
+		}
+	}
+
+	if baseTotalAmount, ok := valuePerAsset[baseAssetKey]; ok {
+		if baseTotalAmount < MinFee {
+			log.Warnf(
+				"min base deposit is %v please top up with another depositfee operation",
+				MinFee,
+			)
+		} else {
+			log.Infof(
+				"base asset %v funded with value %v",
+				baseAssetKey,
+				baseTotalAmount,
+			)
+		}
+	}
+
+	return valuePerAsset[baseAssetKey], unspents, nil
+}
+
+func fetchFeeAccountAddresses(
+	numOfAddresses int,
+	client pboperator.OperatorClient,
+) ([]string, error) {
 	resp, err := client.DepositFeeAccount(
 		context.Background(), &pboperator.DepositFeeAccountRequest{
-			NumOfAddresses: numOfAddresses,
+			NumOfAddresses: int64(numOfAddresses),
 		},
 	)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	printRespJSON(resp)
+	addresses := make([]string, 0, len(resp.GetAddressWithBlindingKey()))
+	for _, v := range resp.GetAddressWithBlindingKey() {
+		addresses = append(addresses, v.Address)
+	}
 
-	return nil
+	return addresses, nil
+}
+
+func estimateFees(numIns, numOuts int) uint64 {
+	ins := make([]int, numIns, numIns)
+	for i := 0; i < numIns; i++ {
+		ins[i] = wallet.P2WPKH
+	}
+
+	outs := make([]int, numOuts, numOuts)
+	for i := 0; i < numOuts; i++ {
+		outs[i] = wallet.P2WPKH
+	}
+
+	size := wallet.EstimateTxSize(ins, nil, nil, outs, nil)
+	return uint64(float64(size) * 0.1)
+}
+
+func waitUntilTxConfirmed(explorerSvc explorer.Service, txid string) error {
+	for {
+		isConfirmed, err := explorerSvc.IsTransactionConfirmed(txid)
+		if err != nil {
+			return err
+		}
+		if isConfirmed {
+			return nil
+		}
+		sleepTime := 20 * time.Second
+		time.Sleep(sleepTime)
+	}
+}
+
+func createOutpoints(txid string, numOuts int) []*pboperator.TxOutpoint {
+	outpoints := make([]*pboperator.TxOutpoint, 0, numOuts)
+	for i := 0; i < numOuts; i++ {
+		outpoints = append(outpoints, &pboperator.TxOutpoint{
+			Hash:  txid,
+			Index: int32(i),
+		})
+	}
+	return outpoints
+}
+
+func getUnspents(explorerSvc explorer.Service, w *trade.Wallet, txid string) ([]explorer.Utxo, error) {
+	tx, err := explorerSvc.GetTransaction(txid)
+	if err != nil {
+		return nil, err
+	}
+
+	utxos := make([]explorer.Utxo, 0)
+	_, script := w.Script()
+	for i, out := range tx.Outputs() {
+		if bytes.Equal(out.Script, script) {
+			revealed, _ := transactionutil.UnblindOutput(out, w.BlindingKey())
+			var valueCommitment, assetCommitment string
+			if out.IsConfidential() {
+				valueCommitment = hex.EncodeToString(out.Value)
+				assetCommitment = hex.EncodeToString(out.Asset)
+			}
+
+			utxos = append(utxos, esplora.NewWitnessUtxo(
+				txid,
+				uint32(i),
+				revealed.Value,
+				revealed.AssetHash,
+				valueCommitment,
+				assetCommitment,
+				revealed.ValueBlinder,
+				revealed.AssetBlinder,
+				script,
+				out.Nonce,
+				out.RangeProof,
+				out.SurjectionProof,
+				tx.Confirmed(),
+			))
+		}
+	}
+
+	return utxos, nil
+}
+
+func waitForOperatorFunds() []string {
```

Comment:
it happens that if the txid is not found (slow to appear on mempool) and the user enter right away the txid it will say `tdex] Transaction not found` and exit the process

Reply:
if the user call again depositfee it will prompt a different address

Reply:
bets way would be to start a ticker to go check that txid until it founds, maybe with a reasonable timeout (1 minute or so)